apps/files/src/components/FileLinkSidebar.vue

```diff
@@ -0,0 +1,263 @@
+      const expireDate = this.capabilities.files_sharing.public.expire_date
+
+      return {
+        enabled: !!expireDate.enabled,
+        days: (expireDate.days) ? expireDate.days : false,
+        enforced: expireDate.enforced === '1'
+      }
+    },
+    $_tooltipTextLink () {
+      return `title: ${this.$gettext('Click to open the link')}; pos: bottom`
+    },
+    $_addButtonLabel () {
+      return this.$gettext('Add public link')
+    },
+    $_deleteButtonLabel () {
+      return this.$gettext('Delete public link')
+    },
+    $_editButtonLabel () {
+      return this.$gettext('Edit public link')
+    },
+    $_copyButtonLabel () {
+      return this.$gettext('Copy link url')
+    },
+    $_copyPrivateLinkLabel () {
+      return this.$gettext('Copy Private Link')
+    },
+    $_privateLinkOfHighlightedFile () {
+      if (!this.highlightedFile) {
+        return false
+      }
+      if (this.highlightedFile.isMounted()) {
+        const file = encodeURIComponent(this.highlightedFile.name)
+        return window.location.href.split('?')[0] + `?scrollTo=${file}`
+      }
+      return this.highlightedFile.privateLink
+    }
+  },
+  methods: {
+    ...mapActions('Files', ['loadLinks', 'purgeLinks', 'removeLink']),
+    $_resetData () {
+      this.params = {
+        id: null,
+        name: this.capabilities.files_sharing.public.defaultPublicLinkShareName,
+        permissions: 1,
+        hasPassword: false,
+        expireDate: (this.$_expirationDate.days) ? moment().add(this.$_expirationDate.days, 'days').endOf('day').toISOString() : null
+      }
+    },
+    $_removePublicLink (link) {
+      this.removeLink({
+        client: this.$client,
+        id: link.id
+      })
+    },
+    $_editPublicLink (link) {
+      this.params = {
+        id: link.id,
+        name: link.name,
+        permissions: parseInt(link.permissions),
+        hasPassword: link.password,
+        expireDate: (link.expiration !== null) ? moment(link.expiration).endOf('day').toISOString() : null
+      }
+      this.visiblePanel = PANEL_EDIT
+    },
+    $_addPublicLink () {
+      this.$_resetData()
+      this.visiblePanel = PANEL_EDIT
+    },
+    $_copyToClipboard (link) {
+      this.$copyText(link.url)
+
+      const clone = event.currentTarget.firstElementChild.cloneNode(true)
+      clone.classList.add('_clipButton')
+      event.currentTarget.append(clone)
+
+      // Remove clone after animation ends
+      setTimeout(() => clone.remove(), 500)
+    },
+    $_clipboardSuccessHandler () {
+      this.linkCopied = true
+
+      // Use copy icon after some time
+      setTimeout(() => {
+        this.linkCopied = false
+      }, 1000)
+    },
+    $_showList () {
+      this.visiblePanel = PANEL_SHOW
+    }
+  }
+}
+</script>
+<style scoped>
+  ._clipButton {
+    animation-name: _clipButton;
+    animation-duration: .5s;
+    animation-timing-function: ease-out;
+    animation-fill-mode: both;
+  }
+
+  @keyframes _clipButton {
```

Comment:
add "TODO" to move these to the design system somehow

Reply:
done